core/templates/dev/head/domain/story_viewer/story-viewer-backend-api.service.ts

```diff
@@ -16,59 +16,79 @@
+  'domain/utilities/url-interpolation.service';
 
-      $http.get(storyDataUrl).then(function(response) {
-        storyDataDict = angular.copy(response.data);
-        if (successCallback) {
-          successCallback(storyDataDict);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
-    };
+@Injectable({
+  providedIn: 'root'
+})
+export class StoryViewerBackendApiService {
+  // TODO(#7176): Replace 'any' with the exact type. This has been kept as
+  // 'any' because 'storyDataDict' is a dict with underscore_cased
+  // keys which give tslint errors against underscore_casing in favor of
+  // camelCasing.
+  storyDataDict: any = null;
+
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private http: HttpClient
+  ) {}
 
-    var _recordChapterCompletion = function(
-        storyId, nodeId, successCallback, errorCallback) {
-      var chapterCompletionUrl = UrlInterpolationService.interpolateUrl(
-        STORY_PROGRESS_URL_TEMPLATE, {
-          story_id: storyId,
-          node_id: nodeId
-        });
-      $http.post(chapterCompletionUrl).then(function(response) {
-        successCallback({
-          summaries: response.data.summaries,
-          nextNodeId: response.data.next_node_id,
-          readyForReviewTest: response.data.ready_for_review_test});
+  _fetchStoryData(storyId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    let storyDataUrl = this.urlInterpolationService.interpolateUrl(
+      StoryViewerDomainConstants.STORY_DATA_URL_TEMPLATE, {
+        story_id: storyId
       });
-    };
 
-    return {
-      fetchStoryData: function(storyId) {
-        return $q(function(resolve, reject) {
-          _fetchStoryData(storyId, resolve, reject);
-        });
-      },
-      recordChapterCompletion: function(storyId, nodeId) {
-        return $q(function(resolve, reject) {
-          _recordChapterCompletion(storyId, nodeId, resolve, reject);
-        });
+    this.http.get(storyDataUrl).toPromise().then((data) => {
+      this.storyDataDict = angular.copy(data);
+      if (successCallback) {
+        successCallback(this.storyDataDict);
+      }
+    }, (error) => {
+      if (errorCallback) {
+        errorCallback(error);
       }
-    };
+    });
+  }
+
+  _recordChapterCompletion(storyId: string, nodeId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    let chapterCompletionUrl = this.urlInterpolationService.interpolateUrl(
+      StoryViewerDomainConstants.STORY_PROGRESS_URL_TEMPLATE, {
+        story_id: storyId,
+        node_id: nodeId
+      });
+    this.http.post(chapterCompletionUrl, {}).toPromise().then((data: any) => {
+      successCallback({
+        summaries: data.summaries,
```

Comment:
How did you get to the point of changing this to `data.summaries` in place of `response.data.summaries`?

Reply:
I noticed Angular HttpClient returns response.data instead of response. I think you have also encountered this as you mentioned [here](https://docs.google.com/document/d/1UfUGhaMXNaVQGuW9GwIl2mlkzVm7QsbldJWOeQcB6kE/)

Reply:
Yes, you're right I mentioned that. I'm curious how you tested this?

Reply:
I tested the chapter completion here, and it's working locally.